admob/app/src/main/java/com/google/firebase/example/admob/MainActivity.java

```diff
@@ -1,19 +1,134 @@
 package com.google.firebase.example.admob;
 
+import android.content.Context;
 import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+
 import androidx.appcompat.app.AppCompatActivity;
 
+import com.google.android.gms.ads.AdListener;
+import com.google.android.gms.ads.AdRequest;
+import com.google.android.gms.ads.AdView;
+import com.google.android.gms.ads.InterstitialAd;
+import com.google.android.gms.ads.LoadAdError;
 import com.google.android.gms.ads.MobileAds;
 
+import devrel.firebase.google.com.firebaseoptions.R;
+
 public class MainActivity extends AppCompatActivity {
 
+    private static final String TAG = "MainActivity";
+
+    private AdView mAdView;
+    private InterstitialAd mInterstitialAd;
+    private Button mLoadInterstitialButton;
+    private Context context = this;
+
     // [START ads_on_create]
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         // ...
         MobileAds.initialize(this);
+        // [END ads_on_create]
+
+        // Initialize the Google Mobile Ads SDK
+        MobileAds.initialize(context);
+
+        // [START load_banner_ad]
+        AdRequest adRequest = new AdRequest.Builder().build();
```

Comment:
Is there a reason all of these snippets are in the `onCreate` function instead of in helper functions?  Having most snippets live in their own function is generally easier to maintain, especially when they don't need to refer to each other's results/variables.

Reply:
Sounds good. And that actually leaves our onCreate snippet untouched.
I've update the PR.